js/src/components/ConnectGoogleSearchConsole.js

```diff
@@ -63,45 +64,97 @@ class ConnectGoogleSearchConsole extends React.Component {
 			"toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=yes, resizable=no, " +
 			"copyhistory=no, width=" + w + ", height=" + h + ", top=" + top + ", left=" + left
 		);
-	}
+	};
 
 	/**
 	 * Saves the authorization code.
 	 *
 	 * @returns {void}
 	 */
 	saveAuthCode() {
-		jQuery.post(
-			yoastWizardConfig.ajaxurl,
-			{
-				action: "wpseo_save_auth_code",
-				ajax_nonce: yoastWizardConfig.gscNonce,
-				authorization: jQuery( "#gsc_authorization_code" ).val(),
-			},
-			this.setProfileList.bind( this ),
-			"json"
-		);
+		let url = yoastWizardConfig.ajaxurl;
+		let config = {
+			action: "wpseo_save_auth_code",
+			ajax_nonce: yoastWizardConfig.gscNonce,
+			authorization: jQuery( "#gsc_authorization_code" ).val(),
+		};
+		let callback = this.setProfileList.bind( this );
+		let dataType = "json";
+
+		this.sendJQueryAJAXrequest( url, config, callback, dataType );
+	}
+
+	/**
+	 * @summary Sends a jQuery AJAX post request.
+	 *
+	 * @param {string} url The URL to post to.
+	 * @param {object} config The parameters to send with the request.
+	 * @param {function} callback The function to call after executing the request.
+	 * @param {string} dataType The dataType for the params.
+	 *
+	 * @returns {void}
+	 */
+	sendJQueryAJAXrequest( url, config, callback, dataType ) {
```

Comment:
Please rename this into `sendJSONRequest` and remove the parameter `dataType` because you are using this function only for JSON.